### PR TITLE
Coordinator test refactor [2/2]

### PR DESCRIPTION
**Pull request by mzhu25**
## Description

Part 2 of #2273 
Utilizes `DeploymentManager`, actor mixins, and `BalanceStore` in the Coordinator integration tests. 
Also implements https://github.com/0xProject/0x-monorepo/pull/2273#issuecomment-543991490 bc I think it's p cool
## Testing instructions

`@PKG=contracts-integrations yarn test`

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/test/actors/fee_recipient.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { ApprovalFactory, SignedCoordinatorApproval } from '@0x/contracts-coordinator';
+import { SignatureType, SignedZeroExTransaction } from '@0x/types';
+
+import { Actor, ActorConfig, Constructor } from './base';
+
+export interface FeeRecipientConfig extends ActorConfig {
+    verifyingContract?: BaseContract;
+}
+
+export function FeeRecipientMixin<TBase extends Constructor>(Base: TBase) {
```
Comment (dorothy-zbornak) [inline contracts/integrations/test/actors/fee_recipient.ts]: Why is this a mixin/decorator? Seems to impose very specific requirements on constructor args *and* the `TBase` type (which should be `Actor`?). This tight coupling smells like a decorator is the wrong pattern here. I kind of feel like this should just be a simple derived class of `Actor` like:

```ts
export class FeeRecipient extends Actor {
    constructor(cfg: FeeRecipientConfig) {
      super(cfg);
      ...
   }
}
````

File: contracts/integrations/test/deployment/deployment_mananger.ts [hunk 2]
```diff
@@ -122,7 +122,9 @@ export interface DeploymentOptions {
 }
 
 export class DeploymentManager {
-    public static protocolFeeMultiplier = new BigNumber(150000);
+    public static readonly protocolFeeMultiplier = new BigNumber(150000);
+    public static readonly gasPrice = new BigNumber(1000000000); // 1 Gwei
```
Comment (dorothy-zbornak) [inline contracts/integrations/test/deployment/deployment_mananger.ts]:
```suggestion
    public static readonly gasPrice = new BigNumber(1e9); // 1 Gwei
```

File: contracts/integrations/test/actors/fee_recipient.ts (hunk 1 above)
Comment (mzhu25) [reply contracts/integrations/test/actors/fee_recipient.ts]: idk if you've seen the mixins in #2273 but the motivation is to be able to mix e.g. `Maker` and `PoolOperator` together. `FeeRecipient` will probably be less commonly used than some of the others, but one could imagine a test in which you'd want a maker that also signs Coordinator fee recipient approvals. We _could_ make FeeRecipient just derive from `Actor`, but I made it a mixin for consistency

File: contracts/integrations/test/coordinator/coordinator.ts [hunk 3]
```diff
@@ -1,419 +1,457 @@
...
+            // Maker -> Fee Recipient
+            localBalanceStore.transferAsset(
+                maker.address,
+                feeRecipient.address,
+                order.makerFee,
+                order.makerFeeAssetData,
             );
+
+            // Protocol fee
+            if (msgValue.isGreaterThanOrEqualTo(DeploymentManager.protocolFee)) {
```
Comment (jalextowle) [inline contracts/integrations/test/coordinator/coordinator.ts]: 🙏

**Pull request merged**